components/org.wso2.carbon.identity.oauth.dcr/src/test/java/org/wso2/carbon/identity/oauth/dcr/service/DCRManagementServiceTest.java

```diff
@@ -0,0 +1,289 @@
+                new ServiceProvider());
+
+        when(mockRegistrationRequestProfile.getRedirectUris()).thenReturn(redirectUris);
+
+        mockOAuthAdminService = mock(OAuthAdminService.class);
+
+        OAuthConsumerAppDTO oAuthConsumerApp = new OAuthConsumerAppDTO();
+        oAuthConsumerApp.setApplicationName(applicationName);
+        oAuthConsumerApp.setOAuthVersion(OAUTH_VERSION);
+        oAuthConsumerApp.setCallbackUrl("dummyCallback");
+
+        if (Integer.valueOf(listSize) == 1) {
+            dummyGrantTypes.clear();
+            dummyGrantTypes.add("code");
+            oAuthConsumerApp.setGrantTypes(dummyGrantTypes.get(0));
+        } else if (Integer.valueOf(listSize) == 2) {
+            dummyGrantTypes.clear();
+            dummyGrantTypes.add("authorization_code");
+            dummyGrantTypes.add("implicit");
+            oAuthConsumerApp.setGrantTypes(dummyGrantTypes.get(0) + " " + dummyGrantTypes.get(1));
+        } else {
+            dummyGrantTypes.clear();
+        }
+
+        oAuthConsumerApp.setOauthConsumerSecret(oauthConsumerSecret);
+
+        whenNew(OAuthAdminService.class).withNoArguments().thenReturn(mockOAuthAdminService);
+
+        when(mockOAuthAdminService
+                .getOAuthApplicationDataByAppName(applicationName)).thenReturn(oAuthConsumerApp);
+
+        RegistrationResponseProfile registrationRequestProfile = dcrManagementService.registerOAuthApplication
+                (mockRegistrationRequestProfile);
+        assertEquals(registrationRequestProfile.getGrantTypes(), dummyGrantTypes);
+        assertEquals(registrationRequestProfile.getClientName(), applicationName);
+    }
+
+    @ObjectFactory
+    public IObjectFactory getObjectFactory() {
+
+        return new org.powermock.modules.testng.PowerMockObjectFactory();
+    }
+
+    private void registerOAuthApplication1() {
+        mockRegistrationRequestProfile = mock(RegistrationRequestProfile.class);
+        when(mockRegistrationRequestProfile.getClientName()).thenReturn(applicationName);
+        when(mockRegistrationRequestProfile.getOwner()).thenReturn(ownerName);
+        when(mockRegistrationRequestProfile.getClientName()).thenReturn(clientName);
+        when(mockRegistrationRequestProfile.getGrantTypes()).thenReturn(dummyGrantTypes);
+        when(mockRegistrationRequestProfile.getTenantDomain()).thenReturn(tenantDomain);
+        applicationName = mockRegistrationRequestProfile.getOwner() + "_" + mockRegistrationRequestProfile
+                .getClientName();
+
+        String carbonHome = Paths.get(System.getProperty("user.dir"), "target").toString();
+        System.setProperty(CarbonBaseConstants.CARBON_HOME, carbonHome);
+        PrivilegedCarbonContext.startTenantFlow();
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain);
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().setUsername(userName);
```

Comment:
Use a different method to start tenant flow, it doesn't match the method name :)

Reply:
added a method startTenantFlow